src/openms/source/ANALYSIS/TARGETED/MetaboTargetedAssay.cpp

```diff
@@ -0,0 +1,498 @@
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Oliver Alka $
+// $Authors: Oliver Alka $
+// --------------------------------------------------------------------------
+
+
+#include <OpenMS/ANALYSIS/TARGETED/MetaboTargetedAssay.h>
+
+#include <OpenMS/COMPARISON/SPECTRA/BinnedSpectrum.h>
+#include <OpenMS/COMPARISON/SPECTRA/BinnedSpectralContrastAngle.h>
+#include <OpenMS/FILTERING/TRANSFORMERS/SpectraMerger.h>
+#include <OpenMS/MATH/MISC/MathFunctions.h>
+
+using namespace OpenMS;
+using namespace std;
+
+namespace OpenMS
+{
+
+  bool MetaboTargetedAssay::intensityLess_(Peak1D a, Peak1D b)
+  {
+    return (a.getIntensity() < b.getIntensity());
+  }
+
+  // method to extract a potential transistions based on the ms/ms based of the highest intensity precursor or a consensus spectrum
+  std::vector <MetaboTargetedAssay> MetaboTargetedAssay::extractMetaboTargetedAssay(const PeakMap& spectra,
+                                                                                    const map<BaseFeature const *, vector < size_t>>& feature_ms2_spectra_map,
+                                                                                    const double& precursor_rt_tol,
+                                                                                    const double& precursor_mz_distance,
+                                                                                    const double& cosine_sim_threshold,
+                                                                                    const double& transition_threshold,
+                                                                                    const bool& method_consensus_spectrum,
+                                                                                    const bool& exclude_ms2_precursor,
+                                                                                    const unsigned int& file_counter)
+  {
+    int transition_group_counter = 0;
+    vector <MetaboTargetedAssay> v_mta;
+
+    for (auto it = feature_ms2_spectra_map.begin();
+              it != feature_ms2_spectra_map.end();
+              ++it)
+      {
+        TargetedExperiment::Compound cmp;
+        cmp.clearMetaInfo();
+
+        vector <ReactionMonitoringTransition> v_rmt;
+
+        String description("UNKNOWN"), sumformula("UNKNOWN"), adduct("UNKNOWN");
+        StringList v_description, v_sumformula, v_adduct;
+
+        double feature_rt;
+        const BaseFeature *min_distance_feature = it->first;
+        feature_rt = min_distance_feature->getRT();
+
+        // extract metadata from featureXML
+        if (!(min_distance_feature-> getPeptideIdentifications().empty()) && !(min_distance_feature->getPeptideIdentifications()[0].getHits().empty()))
```

Comment:
```suggestion
        auto metaboliteIdentifications = min_distance_feature-> getPeptideIdentifications();
        if (!(metaboliteIdentifications.empty()) && !(metaboliteIdentifications[0].getHits().empty()))
```

then its more clear downstream what you are working with and it makes more intuitive sense